api/app/utils/emailer.py

```diff
@@ -22,43 +22,49 @@ def send_mail(recipient, subject, body_text, body_html='', charset='UTF-8', mail
         e -- [description]
     '''
     LOGGER.debug("Entered mailer")
-    if mail_type == 'AMZ':
-        configuration_set = 'ConfigSet'
-        try:
-            LOGGER.info("Building Mimemultipart msg")
-            msg = MIMEMultipart()
-            LOGGER.info("Setting subject %s"%(subject,))
-            msg['Subject'] = subject
-            LOGGER.info("Setting Sender name : %s and Sender email %s"%(SMTP_SENDER_NAME, SMTP_SENDER_EMAIL))
-            msg['From'] = email.utils.formataddr((SMTP_SENDER_NAME, SMTP_SENDER_EMAIL))
-            LOGGER.info("Setting recipient %s"%(recipient))
-            msg['To'] = recipient
-            #msg.add_header('X-SES-CONFIGURATION-SET', configuration_set)
+    if (not DEBUG):
```

Comment:
Normally we'd do this by having an AmazonMailer and a MockMailer, both implementing the common send_mail interface and pass the appropriate one to the controllers (dependency injection), but that's a fair bit more complex, so this is fine for now!

Reply:
How would that piece of logic work? Where would you pass the mailer instance into the controller, and how would you know which one to pass?

Reply:
Just adding to @avishkar58 thought process, the dependency injection can happen using various things, for example as simple as environment variables, where we can base the injection based on a value being set vs unset and inject the appropriate controller. We do not need it at this point in time, but its something that can be done later.

Reply:
@avishkar58 @nischalhp Great, thanks guys! Will look definitely into it :)